tests/wp-unit/tests/include/Core/Posts/Metabox/DeletePostsByCategoryMetaboxTest.php

```diff
@@ -0,0 +1,726 @@
+		$post1 = $this->factory->post->create( array( 'post_title' => 'post1', 'post_status' => 'publish', 'post_category' => array( $cat1 ) ) );
+		$time = current_time('mysql');
+		// Update post1 publish date.
+		wp_update_post(
+			array (
+			'ID'            => $post1,
+			'post_date'     => $day_post,
+			'post_date_gmt' => $day_post,
+			)
+		);
+
+		$post2 = $this->factory->post->create( array( 'post_title' => 'post2', 'post_status' => 'publish', 'post_category' => array( $cat2 ) ) );
+
+		// Assert that each post status is publish.
+		$post1_status = get_post_status( $post1 );
+		$post2_status = get_post_status( $post2 );
+
+		$this->assertEquals( 'publish', $post1_status );
+		$this->assertEquals( 'publish', $post2_status );
+
+		// Assert that each category has one post.
+		$posts_in_cat1 = $this->get_posts_by_category( $cat1 );
+		$posts_in_cat2 = $this->get_posts_by_category( $cat2 );
+
+		$this->assertEquals( 1, count( $posts_in_cat1 ) );
+		$this->assertEquals( 1, count( $posts_in_cat2 ) );
+
+		// call our method.
+		$delete_options = array(
+			'post_type'      => array( 'post' ),
+			'selected_cats'  => array( $cat1, $cat2 ),
+			'restrict'       => true,
+			'date_op'        => 'before',
+			'days'           => 1,
+			'private'        => false,
+			'limit_to'       => false,
+			'force_delete'   => false,
+		);
+		$this->metabox->delete( $delete_options );
+
+		// Assert that post1 status moved to trash post2 status is publish.
+		$post1_status = get_post_status( $post1 );
+		$post2_status = get_post_status( $post2 );
+
+		$this->assertEquals( 'trash', $post1_status );
+		$this->assertEquals( 'publish', $post2_status );
+
+		// Assert that cat1 has no post and cat2 has one post.
+		$posts_in_cat1 = $this->get_posts_by_category( $cat1 );
+		$posts_in_cat2 = $this->get_posts_by_category( $cat2 );
+
+		$this->assertEquals( 0, count( $posts_in_cat1 ) );
+		$this->assertEquals( 1, count( $posts_in_cat2 ) );
+	}
+
+	/**
+	 * Add tests to test deleting posts that are posted within the last x days.
+	 */
+	public function test_for_delete_posts_last_x_days() {
+		//Set post publish date
+		$day_post = date('Y-m-d H:i:s', strtotime('-2 day'));
+		// Create two categories.
+		$cat1 = $this->factory->category->create( array( 'name' => 'cat1' ) );
+		$cat2 = $this->factory->category->create( array( 'name' => 'cat2' ) );
+
+		// Assign the cat1 and cat2 to post1 and post2
+		$post1 = $this->factory->post->create( array( 'post_title' => 'post1', 'post_status' => 'publish', 'post_category' => array( $cat1 ) ) );
+		$time = current_time('mysql');
+		// Update post1 publish date.
+		wp_update_post(
+			array (
+			'ID'            => $post1,
+			'post_date'     => $day_post,
+			'post_date_gmt' => $day_post,
+			)
+		);
+
+		$post2 = $this->factory->post->create( array( 'post_title' => 'post2', 'post_status' => 'publish', 'post_category' => array( $cat2 ) ) );
+
+		// Assert that each post status is publish.
```

Comment:
Remove this post status check here. It doesn't make sense for testing 'Delete Posts by Category'

Reply:
Fixed